### PR TITLE
add Redux in React components

### PR DESCRIPTION
**Pull request by Nezerwa**
in this branch I :
-[]Connected my store to the main component of your React app.
-[]Defined a state slice which  contain my books
-[]Dispatched actions using useDispatch

### CODE REVIEW

**Review by levy002 — COMMENTED** [review]
## Hello @Nezerwa 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request :red_circle: :red_circle: 

You have submitted a project, that fails linters. Kindly make sure to solve linters again and submit your project again.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/App.js [hunk 1]
```diff
@@ -1,23 +1,24 @@
...
+import CategoriesPage from './pages/CategoriesPage';
+
+const router = createBrowserRouter(
+  createRoutesFromElements(
+    <Route path="/" element={<RootLayout />}>
+      <Route index element={<BooksPage />} />
+      <Route path="categories" element={<CategoriesPage />} />
+    </Route>,
+  ),
+);
```
Comment (andersonlebon) [inline src/App.js]: image

- [ ] Kindy solve these errors when your project starts,  they are coming `react-router-dom` you are using some modules that are not found in `react-router-dom`. kindly solve this.

File: src/pages/BooksPage.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      <div className="books-list-container row">
+        <ul className="books-list">
+          {books.map((bk) => (
+            <Book key={bk.id} title={bk.title} author={bk.author} id={bk.id} />
+          ))}
+        </ul>
+      </div>
+      <Form />
+    </div>
+  );
```
Comment (andersonlebon) [inline src/pages/BooksPage.js]: - [ ] Kindly you need to use the `react-redux` and retrieve books from the redux store. With this whenever the state of redux changes your content should change also. for example

```.jsx
import React from 'react'
import { useSelector } from 'react-redux'

function BookList() {
    const books = useSelector((state) => state.books.books)

    return (
        <div>{/* Booklist JSX */}</div>
    )
}

export default BookList
```

File: src/components/Book.js [hunk 3]
```diff
@@ -1,22 +1,26 @@
...
+  const { title, author, id } = props;
+  return (
+    <div>
+      <h2>{title}</h2>
+      <span>
+        By
+        {author}
+      </span>
+      <button type="button" data-id={id}>
+        Remove
```
Comment (andersonlebon) [inline src/components/Book.js]: - [ ] Kindly make sure that when this button is clicked an action is dispatched and the book is removed from the redux store. like this example below 👇 
```.jsx
// Note that this is just an example you can address this as you want but make sure that book is removed
//from the redux store using `react-redux`(useDispatch)

 import React from "react"
import { useDispatch } from "react-redux"
import { removeBook } from "path/to/booksSlice"

function Book({ id, title, author }) {
  const dispatch = useDispatch()

  return (
    <div>
      {/* Rest of Book JSX */}
      <Button onClick={() => dispatch(removeBook(id))}>Remove</Button>
    </div>
  )
}

export default Book
   

```

File: src/components/Form.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+          aria-label="Book author input"
+          onChange={authorChangeHandler}
+        />
+        <button type="submit">Submit</button>
+      </form>
+    </div>
+  );
+}
+
+export default Form;
```
Comment (andersonlebon) [inline src/components/Form.js]: - [ ] Kindly make sure that when this form is submitted an action is dispatched and the book is saved in the redux store. like this example below 👇 
```.jsx
// Note that this is just an example you can address this as you want but make sure that book is saved in the
// redux store using `react-redux`(useDispatch)



import React from "react"
import { useDispatch } from "react-redux"
import { addBook } from "path/to/booksSlice"

function NewBookForm() {
  const [title, setTitle] = useState("")
  const [author, setAuthor] = useState("")
  const books = useSelector((state) => state.books.books)
  const dispatch = useDispatch()

  // Other logic that controls form state

  return (
    <div>
      {/* Rest of form JSX */}
      <Button onClick={() => dispatch(addBook({ title, author }))}>
        Add book
      </Button>
    </div>
  )
}

export default NewBookForm
```

File: src/pages/CategoriesPage.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  return (
+    <div>
+      <div className="categories-wrapper row">
+        <button type="button">Check status</button>
+      </div>
+    </div>
+  );
+}
+
+export default CategoriesPage;
```
Comment (andersonlebon) [inline src/pages/CategoriesPage.js]: - [ ] Kindly make sure that when the check status button is clicked, an action is dispatched in the redux store. like this example below 👇 

```.jsx

import React from "react"
import { useDispatch } from "react-redux"
import { checkStatus } from "path/to/categoriesSlice"

function Categories() {
  const dispatch = useDispatch()

  return (
    <div>
      <Button onClick={() => dispatch(checkStatus("Under construction"))}>
        Check status
      </Button>
      {/* Rest of Categories JSX */}
    </div>
  )
}

export default Categories
```

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @Nezerwa 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there



## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question `@andersonlebon` so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js (hunk 1 above)
Comment (Abenezer-Tilahun) [reply src/App.js]: Fixed.

File: src/pages/BooksPage.js (hunk 2 above)
Comment (Abenezer-Tilahun) [reply src/pages/BooksPage.js]: Fixed.

File: src/components/Book.js (hunk 3 above)
Comment (Abenezer-Tilahun) [reply src/components/Book.js]: Fixed.

File: src/components/Form.js (hunk 4 above)
Comment (Abenezer-Tilahun) [reply src/components/Form.js]: Fixed.

File: src/pages/CategoriesPage.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
```
Comment (Abenezer-Tilahun) [inline src/pages/CategoriesPage.js:13]: - [ Optional ]   As suggested by the previous reviser Kindly make sure that when the check status button is clicked, an action is dispatched in the redux store. like this example below 👇

```
import React from "react"
import { useDispatch } from "react-redux"
import { checkStatus } from "path/to/categoriesSlice"

function Categories() {
  const dispatch = useDispatch()

  return (
    <div>
      <Button onClick={() => dispatch(checkStatus("Under construction"))}>
        Check status
      </Button>
      {/* Rest of Categories JSX */}
    </div>
  )
}

export default Categories
```

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @Nezerwa 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.

## Optional suggestions 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**